charts/opentelemetry-collector/values.yaml

```diff
@@ -8,6 +8,11 @@ fullnameOverride: ""
 # Valid values are "daemonset" and "deployment".
 mode: ""
 
+# When true, the chart will not use .Values.config and will not provide the collector a configuration.
+# The user is responsible for providing a configmap with the collector's configuration.
+# See the "deployment-use-existing-configmap" example for values.yaml details.
+useExistingConfigMap: false
```

Comment:
Should we follow the same approach that we have for `serviceAccount` and ` clusterRole` and use the following interface instead:

```yaml
configMap:
  # Specifies whether a configMap should be created
  create: false
  # The name of the configMap to use.
  # If not set and create is true, a name is generated using the chart name.
  name: ""
```

Reply:
I don't get why `Command Args, extraVolumes, and extraVolumeMounts` are related to the `configMap` setting

Reply:
In this situation if we don't generate the configmap ourselves and the user wants to provide their own they have to set some entries in those 3 fields, like in the new example: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/098d0712f3006ce122004b6c67f471a90f5b788f/charts/opentelemetry-collector/examples/deployment-use-existing-configMap/deployment-values.yaml.

Reply:
What kind of configuration interface do you envision?

Reply:
@dmitryax what do you think?

Reply:
Yeah, this sounds good to me. Just to avoid any imlicit magic

Reply:
@JaredTan95 Can you update the PR to use this config interface?

Reply:
done